ansible/roles/test/files/helpers/ferret.py

```diff
@@ -179,8 +179,12 @@ def action(self, interface):
         gre_type  = data[0x24:0x26]
 
         gre_type_r = struct.unpack('!H', gre_type)[0]
-        if gre_type_r == 0x88be:   # Broadcom
-            arp_request = data[0x26:]
+        if gre_type_r == 0x88be:   # ERSPAN type 2
+            # TODO: handle GRE optional fields
+            # Ethernet(14) + IP(20) + GRE(4) + ERSPAN(8) = 34 = 0x2e
+            # References: https://tools.ietf.org/html/rfc1701
+            #             https://tools.ietf.org/html/draft-foschiano-erspan-00
+            arp_request = data[0x2E:]
```

Comment:
Have you tested this change?

Reply:
Yes. I have tested these changes on the BFN platform. Will be great if you could try it on another platform.

Reply:
Can you please revert the change and add a new platform which is bfn?

Reply:
I do not understand the reason. 
Different behavior for different vendors for one protocol? 
I leave a comment about the sources. I see that [this document](https://tools.ietf.org/html/draft-foschiano-erspan-00) is just a draft, not RFC. 
Could you point me to the document which describes another behavior for the same GRE type?
Or if I missed something, please explain.

Reply:
Each vendor implemented this feature on it's own. I didn't read any documentation or drafts. I captured packets from working equipment and check offset from the captures. So I'm pretty sure this will work for broadcom (at least it worked for me, when I tested it.)

Reply:
Also **34 = 0x2e** this is wrong.
34 = 0x22
0x2e = 46

Reply:
I believe its typo (14+20+4+8=46=0x2e)

Reply:
Reverted the change and add vendor-specific code?